client/state/sites/hooks/use-premium-global-styles.ts

```diff
@@ -1,45 +1,14 @@
-import { isEnabled } from '@automattic/calypso-config';
 import { select } from '@wordpress/data';
 import { useQuery } from 'react-query';
 import { useSelector } from 'react-redux';
 import { ONBOARD_STORE } from 'calypso/landing/stepper/stores';
-import wpcom from 'calypso/lib/wp';
 import { getSite } from 'calypso/state/sites/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
-
-type GlobalStylesStatus = {
-	shouldLimitGlobalStyles: boolean;
-	globalStylesInUse: boolean;
-};
-
-// While we are loading the Global Styles Info we can't assume that we should limit global styles, or we would be
-// showing notices for paid sites until we fetch the data from the server.
-const DEFAULT_GLOBAL_STYLES_INFO: GlobalStylesStatus = {
-	shouldLimitGlobalStyles: false,
-	globalStylesInUse: false,
-};
-
-const getGlobalStylesInfoForSite = ( siteId: number | null ): GlobalStylesStatus => {
```

Comment:
The reason for moving these functions in a new file `use-site-global-styles-status.ts` is so that we can reuse this query hook without introducing dependency of the `ONBOARD_STORE` which we don't need for the Theme Showcase.

Reply:
+1, feels a bit odd to have a strong dependency to the `ONBOARD_STORE within the hook. Let's have a separate issue for this!  #71653